gcc/rust/ast/rust-ast-dump.cc

```diff
@@ -746,7 +745,24 @@ Dump::visit (Function &function)
 
 void
 Dump::visit (TypeAlias &type_alias)
-{}
+{
+  // TODO does this represent associated type
+  //  - if so type aliased is optional and TypeParamBounds are legal
+  // https://doc.rust-lang.org/reference/items/associated-items.html#associated-types
+  // Visibility? type IDENTIFIER GenericParams? ( : TypeParamBounds )?
+  // WhereClause? ( = Type)? ;
+
+  if (type_alias.has_visibility ())
+    emit_visibility (type_alias.get_visibility ());
+  stream << "type " << type_alias.get_new_type_name ();
+  if (type_alias.has_generics ())
+    emit_generic_params (type_alias.get_generic_params ());
+  if (type_alias.has_where_clause ())
+    {} // FIXME: WhereClause
+  stream << " = ";
+  type_alias.get_type_aliased ()->accept_vis (*this);
+  stream << ";\n\n";
```

Comment:
Very tiny nitpick: Why two newlines? Is there a specific reason?

No need to fix this right now, I think it's completely fine to leave it that way. It produces valid Rust code and that's what matters :)

Reply:
I had an example for that but it actually does not make sense in the general case. Fixed.